lib/matplotlib/style/core.py

```diff
@@ -244,17 +248,25 @@ def update_nested_dict(main_dict, new_dict):
     return main_dict
 
 
+class _StyleLibrary(dict):
+    def __getitem__(self, key):
+        if key in _DEPRECATED_SEABORN_STYLES:
+            _api.warn_deprecated("3.6", message=_DEPRECATED_SEABORN_MSG)
+            key = _DEPRECATED_SEABORN_STYLES[key]
+
+        return dict.__getitem__(self, key)
+
+
 # Load style library
 # ==================
 _base_library = read_style_directory(BASE_LIBRARY_PATH)
-library = None
+library = _StyleLibrary()
 available = []
 
 
 def reload_library():
     """Reload the style library."""
-    global library
-    library = update_user_library(_base_library)
+    library.update(update_user_library(_base_library))
```

Comment:
To match previous semantics, library should be cleared first?

Reply:
Good catch.